Detectors/GlobalTracking/src/MatchHMP.cxx

```diff
@@ -258,7 +259,24 @@ void MatchHMP::addConstrainedSeed(o2::track::TrackParCov& trc, o2::dataformats::
     mTracksLblWork[o2::globaltracking::MatchHMP::trackType::CONSTR].emplace_back(mRecoCont->getTPCITSTrackMCLabel(srcGID));
   }
 
-  mTracksIndexCache[o2::globaltracking::MatchHMP::trackType::CONSTR].push_back(it);
```

Comment:
Why do you need to cache the track into the mTracksWorkmTracksWork and mTrackGid and only then to apply fiducial selection? Why not caching only the tracks which can be used?

Reply:
Yes, you are right, I will do like that!

Reply:
still don't understand why do you need to push to `mTracksIndexCache` if the track is not used. Now you are adding the same `it` multiple times and then loop over these multiple entries. I think mTracksIndexCache also should be updated only when the track is accepted.